README.md

```diff
@@ -28,10 +26,132 @@ To generate organization-wide token:
+  state: ready
+```
+where:
+  - `push-robot-account` is the name of quay robot account in the configured quay organization with write premissions to the repository.
+  - `push-secret` is a `Secret` of dockerconfigjson type that contains image repository access token with write permissions.
+
+### User defined image repository name
+
+One may request custom image repository name by setting `spec.image.name` field on the `ImageRepository` object creation.
+Note, it's not possible to change image repository name after creation.
+Any changes to the field will be reverted by the operator.
+
+---
+**NOTE**
+
+Image repository name is always prefixed with the `ImageRepository` namespace plus `/`.
+
+---
+
+### Image repository visibility
+
+It's possible to control image repository visibility by `spec.image.visibility` field.
+Allowed values are:
+ - `public` (default)
+ - `private`
+
+It's possible to set visibility on creation or any time later.
+
+---
+**NOTE**
+
+Your quay.io organization plan should allow creation of private repositories.
+In case your quay.io organization has free plan that does not allow setting repositories as private, then if `private` visibility was requested:
+ - on `ImageRepository` creation, then the creation will fail.
+ - after `ImageRepository` creation, then `status.message` will be set and `spec.image.visibility` reverted back to `public`.
+
+---
+
+### Credentials rotation
+
+It's possible to request robot account token rotation by adding:
+```yaml
+...
+spec:
+  ...
+  credentials:
+    regenerate-token: true
+  ...
+```
+After token rotation, the `spec.credentials` section will be deleted and `status.credentials.generationTimestamp` updated.
+
+### Error handling
+
+If a critical error happens on image repository creation, then `status.state` is set to `failed` along with `status.message` field.
+To retry image repository provision, one should recreate `ImageRepository` object.
+
+If a non critical error happens, then `status,message` is set and corresponding `spec` fields are reverted.
+
+---
+**NOTE**
+
+After any successful operation, `status.message` is cleared.
+
+---
+
+## AppStudio Component image repository
+
+There is a special use-case for image repository that stores user's `Component` built images.
+
+To request image repository provision for the `Component`'s builds, the following labels must be added on `ImageRepository` creation:
+ - `appstudio.redhat.com/component` with the `Component` name as the value
+ - `appstudio.redhat.com/application` with the `Application` name to which the `Component` belongs to.
+
+---
+**NOTE**
+
+Described above labels must be added on `ImageRepository` object creation.
+Adding them later will have no effect.
+
+---
+
+The key difference from general purpose workflow are:
+ - second robot account and corresponding secret created with read (pull) only access to the image repository.
+ - the pull secret is propagated into all `Application` environments via `RemoteSecret`.
```

Comment:
We know that the "environments" means individual deployment environment, staging and prod. Do we need to mention it here explicitly?

Is "Application environments" described in some other document? If yes, it should be good to link to it.

Reply:
I looked briefly and failed to find the definition. The closest mention of it is [here](https://github.com/redhat-appstudio/book/blob/ce0a4129bde41db0008b821d10517fca79480014/book/release-service.md?plain=1#L46). Don't think we should include the link above here.